tls/s2n_signature_algorithms.c

```diff
@@ -149,8 +149,9 @@ int s2n_get_and_validate_negotiated_signature_scheme(struct s2n_connection *conn
     /* We require an exact match in TLS 1.3, but all previous versions can fall back to the default SignatureScheme.
      * This means that an s2n client will accept the default SignatureScheme from a TLS server, even if the client did
      * not send it in it's ClientHello. This pre-TLS1.3 behavior is an intentional choice to maximize support. */
-    struct s2n_signature_scheme default_scheme;
-    POSIX_GUARD(s2n_choose_default_sig_scheme(conn, &default_scheme));
+    s2n_mode peer_mode = (conn->mode + 1) % 2;
```

Comment:
Could this be a macro? I feel like I've needed this behavior before and also this is weird looking as it is.